### DISCUSSION TITLE
BV20L NanoEls

### DISCUSSION THREAD

**Discussion by Siroblong2021** [kachurovskiy/nanoels#47] [Show and tell]
Hello there,
I have now the hardware mounted and running on the BV20L
Encoder seems to work and show correct on the Display, maybe the factor needs again to be changed, i ll test this tomorrow.
The wiring still needs some study, since i have very few experience on this topic.

So long
![lathe els01](https://user-images.githubusercontent.com/84715670/155236191-786dfbdb-f72e-4189-8b1b-31339ae0bb66.JPG)
![lathe els02](https://user-images.githubusercontent.com/84715670/155236289-f6af75e8-55ae-4c96-8564-afdd9784244f.JPG)
![lathe els03](https://user-images.githubusercontent.com/84715670/155236295-67c97fe0-6c93-46d7-ac18-4bab85195542.JPG)
![lathe els04](https://user-images.githubusercontent.com/84715670/155236325-59e0c7be-1a75-4261-a440-5e7540be9ade.JPG)

**Comment by kachurovskiy** [comment]
Hi, nice work! Thanks for sharing.

I too realized that we don't have a good PCB solution for those willing to use the big display and have the following version of NanoEls in the works:

![image](https://user-images.githubusercontent.com/517919/155386205-3db7c958-5d45-4167-ac9c-b995c13098bc.png)

So far I only did the hardware part though and didn't adapt the code or even turned it on, too lazy :-/

**Comment by Siroblong2021** [comment]
Hello, its been a while but now im back on the project.
I have made a few minor changes and the hardware setup is now ready.
My problem now:
When i start the ELS the stepper goes into a blocked mode, when i turn the main spindle the stepper gives a light weeping sound, also when i give a little torque by hand to the leadscrew.
My assumption, possibly:
-Wrong leadscrew direction in the code/How can i check that?
-wrong settings in the stepper controller
-wrong hardware cable connections
-other unknown cause

I d like to know your opinion, 

Greetings

**Reply by kachurovskiy** [reply]
Hey, it's hard to tell, maybe something will get cleared up if you could post the pictures of your driver (toggles state and settings table) and the code you used (https://gist.github.com/)

**Comment by Siroblong2021** [comment]
Hello,
I found now that i made two mistakes, i didnt save the Code since i believed it would be accessible on the Nano, second is that i have no source whatsoever that tells me the Solution of the Encoder, so its unknown.
What is known that it is one of these:100,200,360,400,500,600.
I believe i can find this out myself, IF i have the correct Code on the Nano.
Since i have not enough Experience in Coding i dare to ask you (I really hate this!) to complete the Code with the following Settings :
-Lead screw Pitch: 3MM
-Stepper Motor/Lead screw conn.: Linear, as to be seen- on the Pics
-Big Display
-Encoder Solution, We start with 600, i will then be able to change the Code myself until it fits. 
I added some Pics to give you a better idea of the Setup, the toggle switches are still default i believe.  Greetings,
![DSC05637](https://user-images.githubusercontent.com/84715670/170884487-cf8f11c9-450f-4caf-89bd-87e705305d9d.JPG)
![DSC05638](https://user-images.githubusercontent.com/84715670/170884494-56395c7c-a330-4467-acd0-59a06139d9d7.JPG)
![DSC05640](https://user-images.githubusercontent.com/84715670/170884498-58ca8600-1889-4f11-b430-0cfc011ff4f5.JPG)
![DSC05641](https://user-images.githubusercontent.com/84715670/170884502-83cdc318-bef3-49e4-8e32-e5188e7521ba.JPG)
![DSC05642](https://user-images.githubusercontent.com/84715670/170884508-93e020fc-49d7-4ff0-beb9-b94353c4304b.JPG)
![DSC05643](https://user-images.githubusercontent.com/84715670/170884515-2b8dac26-ca58-4cbd-a9d4-8a1add4e7c9c.JPG)
![DSC05644](https://user-images.githubusercontent.com/84715670/170884519-c0c5d0f0-e3c9-4bbc-b450-4440efdfed14.JPG)
![DSC05645](https://user-images.githubusercontent.com/84715670/170884523-401baf91-eaa3-4ff3-a897-5f2509e798f3.JPG)
![DSC05646](https://user-images.githubusercontent.com/84715670/170884527-f76ca6d7-3372-4544-b613-93c99a028d9e.JPG)

**Comment by Siroblong2021** [comment]
I just received Email from China:
Receiver Resolution:600.
Very nice.

**Comment by kachurovskiy** [comment]
Try the following code: https://gist.github.com/kachurovskiy/3b1f4aaec52c8ddc70e7441250d24507

Set SW1 to OFF to get 800 steps which is recommended.

If leadscrew is moving in the wrong direction, set SW5 to ON.
